cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py

```diff
@@ -133,29 +132,39 @@ def update_kubelet_status():
 
 @when('kubernetes-worker.components.installed', 'kube-api-endpoint.available',
       'tls_client.ca.saved', 'tls_client.client.certificate.saved',
-      'tls_client.client.key.saved', 'kube-dns.available')
-def start_worker(kube_api, kube_dns):
+      'tls_client.client.key.saved', 'kube-dns.available', 'cni.available')
+def start_worker(kube_api, kube_dns, cni):
     ''' Start kubelet using the provided API and DNS info.'''
     servers = get_kube_api_servers(kube_api)
     # Note that the DNS server doesn't necessarily exist at this point. We know
     # what its IP will eventually be, though, so we can go ahead and configure
     # kubelet with that info. This ensures that early pods are configured with
     # the correct DNS even though the server isn't ready yet.
+
     dns = kube_dns.details()
+
     if (data_changed('kube-api-servers', servers) or
             data_changed('kube-dns', dns)):
         # Initialize a FlagManager object to add flags to unit data.
         opts = FlagManager('kubelet')
         # Append the DNS flags + data to the FlagManager object.
-        opts.add('--cluster-dns', dns['sdn-ip'])
+
+        opts.add('--cluster-dns', dns['sdn-ip']) # FIXME: sdn-ip needs a rename
```

Comment:
Is the original code using the wrong value here? Or does the name just need CNI? I am all for renaming it as you see fit.